ouroboros-consensus/src/Ouroboros/Consensus/Node/Types.hs

```diff
@@ -0,0 +1,45 @@
+{-# LANGUAGE DeriveAnyClass     #-}
+{-# LANGUAGE DeriveGeneric      #-}
+{-# LANGUAGE DerivingStrategies #-}
+
+module Ouroboros.Consensus.Node.Types (
+  CandidateDecline (..),
+  CandidateFragment (..),
+  CandidateFingerprint (..),
+  ) where
+
+import           GHC.Generics (Generic)
+import           NoThunks.Class
+
+import           Ouroboros.Network.AnchoredFragment (AnchoredFragment)
+
+import           Ouroboros.Consensus.Block (Point)
+
+-- | What ChainSync provides per peer to BlockFetch.
+data CandidateFragment header = CandidateFragment {
+    candidateIsLowDensity :: !Bool
```

Comment:
Rename this field; something more precise. EG Density ultimately depends on the next header; could even apply to empty chain.